tests/src/ListExtraTests.elm

```diff
@@ -0,0 +1,82 @@
+module ListExtraTests exposing (tests)
+
```

Comment:
What I would suggest here is looking at https://github.com/elm-community/list-extra/blob/6.0.0/tests/Tests.elm and:

* copy the relevant tests
* make whatever the smallest changes are to fit the function signatures here

That is, instead of working from `ListTests.elm` as a base, work from the actual tests for `List.Extra` -- the idea is to prove that we can pass the `List.Extra` tests (modified as little as needed to fit our context).

Then, in a separate test file, you could test anything else you want to test -- i.e. things that `List.Extra` doesn't test, or behaviours that are unique to `DictList`.